programs/vest_api/src/vest_processor.rs

```diff
@@ -67,22 +66,21 @@ fn redeem_tokens(
     vest_state: &mut VestState,
     keyed_accounts: &mut [KeyedAccount],
 ) -> Result<(), VestError> {
-    let date_keyed_account = &keyed_accounts[0];
-    if date_keyed_account.account.owner != solana_config_api::id() {
+    let oracle_keyed_account = &keyed_accounts[0];
+    if oracle_keyed_account.account.owner != solana_config_api::id() {
         return Err(VestError::UnexpectedProgramId);
     }
 
-    if *date_keyed_account.unsigned_key() != vest_state.oracle_pubkey {
+    if *oracle_keyed_account.unsigned_key() != vest_state.oracle_pubkey {
         return Err(VestError::Unauthorized);
     }
 
     if &vest_state.payee_pubkey != keyed_accounts[2].unsigned_key() {
-        trace!("destination missing");
         return Err(VestError::DestinationMissing);
     }
 
     let date_config: DateConfig =
-        deserialize(get_config_data(&date_keyed_account.account.data).unwrap()).unwrap();
+        deserialize(get_config_data(&oracle_keyed_account.account.data).unwrap()).unwrap();
```

Comment:
need to throw an InstructionError here, not unwrap()

Reply:
Fixed. Left the Config unwrap because it's unreachable, but removed the Date unwrap, which is possible if you mix it up with a different Config account.